src/Compilers/CSharp/Test/Syntax/LexicalAndXml/SyntaxTokenParserTests.cs

```diff
@@ -0,0 +1,212 @@
+
+            }
+            """, parser.ParseNextToken());
+
+        AssertToken(expectedKind: SyntaxKind.EndOfFileToken, expectedContextualKind: SyntaxKind.None, "", parser.ParseNextToken());
+        AssertToken(expectedKind: SyntaxKind.EndOfFileToken, expectedContextualKind: SyntaxKind.None, "", parser.ParseNextToken());
+    }
+
+    [Fact]
+    public void DirectiveContextIsPreservedAcrossParseNextToken()
+    {
+        var sourceText = SourceText.From("""
+            #if true
+            class C
+            {
+
+            #else
+            }
+            #endif
+            """);
+        var parser = SyntaxFactory.CreateTokenParser(sourceText, TestOptions.Regular);
+
+        AssertToken(expectedKind: SyntaxKind.ClassKeyword, expectedContextualKind: SyntaxKind.None, """
+            #if true
+            class 
+            """, parser.ParseNextToken());
+
+        AssertToken(expectedKind: SyntaxKind.IdentifierToken, expectedContextualKind: SyntaxKind.None, """
+            C
+
+            """, parser.ParseNextToken());
+
+        AssertToken(expectedKind: SyntaxKind.OpenBraceToken, expectedContextualKind: SyntaxKind.None, """
+            {
+
+            """, parser.ParseNextToken());
+
+        AssertToken(expectedKind: SyntaxKind.EndOfFileToken, expectedContextualKind: SyntaxKind.None, """
+
+            #else
+            }
+            #endif
+            """, parser.ParseNextToken());
+
+        AssertToken(expectedKind: SyntaxKind.EndOfFileToken, expectedContextualKind: SyntaxKind.None, "", parser.ParseNextToken());
+    }
+
+    [Fact]
+    public void SkipForwardTo()
+    {
+        var sourceText = SourceText.From("""
+            This is not C#
+
+            // Hello world
+            class C
+            {
+
+            }
+            """);
+        var parser = SyntaxFactory.CreateTokenParser(sourceText, TestOptions.Regular);
+
+        parser.SkipForwardTo(16);
+
+        AssertToken(expectedKind: SyntaxKind.ClassKeyword, expectedContextualKind: SyntaxKind.None, """
+            // Hello world
+            class 
+            """, parser.ParseNextToken());
+
+        parser.SkipForwardTo(39);
+
+        AssertToken(expectedKind: SyntaxKind.OpenBraceToken, expectedContextualKind: SyntaxKind.None, """
+            {
+
+            """, parser.ParseNextToken());
+
+        AssertToken(expectedKind: SyntaxKind.CloseBraceToken, expectedContextualKind: SyntaxKind.None, """
+
+            }
+            """, parser.ParseNextToken());
+    }
+
+    [Fact]
+    public void SkipForwardTo2()
+    {
+        var sourceText = SourceText.From("""class""");
+        var parser = SyntaxFactory.CreateTokenParser(sourceText, TestOptions.Regular);
+        parser.SkipForwardTo(1);
+
+        AssertToken(expectedKind: SyntaxKind.IdentifierToken, expectedContextualKind: SyntaxKind.None, """lass""", parser.ParseNextToken());
+    }
```

Comment:
test skipping to the end and getting EOF.
test getting multiple EOFs at the end of hte file.
test skipping backwards.

Reply:
> test getting multiple EOFs at the end of hte file.

Already part of `ParseNextToken` and `DirectiveContextIsPreservedAcrossParseNextToken`.

> test doc comment as leading trivia.

What about that specifically?

Reply:
just have a test that shows it properly lexes, and that you get a DocComment trivia back that you can dive into :)